docs/http_api.rst

```diff
@@ -0,0 +1,1092 @@
+    :status 400: Missing required information to create team.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        POST /teams/ HTTP/1.1
+        Content-Type: application/json
+
+        {
+            "name": "Lord Commanders",
+            "organization": 7
+        }
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 201 Created
+        Content-Type: application/json
+
+        {
+            "id": 2,
+            "name": "Lord Commanders",
+            "permissions": [],
+            "url": "https://example.org/teams/2",
+            "organization": 7
+        }
+
+.. http:get:: /teams/(int:team_id)
+
+    Get the details of a team.
+
+    :>json int id: the ID of the team.
+    :>json str url: the URL of the team.
+    :>json str name: the name of the team.
+    :>json int organization: The id of the organization that the team belongs to.
+    :>json list permissions:
+        The permission list of the team. Each permission is an object
+        containing the fields "id", "permission", and "object_id".
+    :status 200: Successfully retrieved team.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        GET /teams/2 HTTP/1.1
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 200 OK
+        Content-Type: application/json
+
+        {
+            "id": 2,
+            "name": "Lord Commanders",
+            "permissions": [],
+            "url": "https://example.org/teams/2",
+            "organization": 7
+        }
+
+.. http:put:: /teams/(int:team_id)
+
+    Update the details of a team.
+
+    :<json str name: The name of the team.
+
+    :>json int id: the id of the updated team.
+    :>json str url: The URL of the updated team.
+    :>json str name: the name of the team.
+    :>json int organization: The id of the organization that the team belongs to.
+    :>json list permissions:
+        The permission list of the team. Each permission is an object
+        containing the fields "id", "permission", and "object_id".
+    :status 200: successfully updated team.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        PUT /teams/2 HTTP/1.1
+        Content-Type: application/json
+
+        {
+            "name": "Brotherhood without banners"
+        }
+
+    **Example reponse**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 200 OK
+        Content-Type: application/json
+
+        {
+            "id": 2,
+            "name": "Brotherhood without banners",
+            "permissions": [],
+            "url": "https://example.org/teams/2",
+            "organization": 7
+        }
+
+.. http:delete:: /teams/(int:team_id)
+
+    Remove a team.
+
+    :status 204: Team successfully deleted.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        DELETE /teams/2 HTTP/1.1
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 204 No Content
+
+.. http:put:: /teams/(int:team_id)/permission
```

Comment:
Yes that's more correct, since we're creating a permission, not adding an existing permission to a team. I'll update it.